### ISSUE TITLE
Support PersistenceContextType.EXTENDED in Quarkus Hibernate

### ISSUE THREAD

**Issue by Treiblesschorle** [quarkusio/quarkus#32940]
### Description

I'm currently migrating a large existing application to Quarkus and I want to use [Quarkus Hibernate ORM](https://quarkus.io/guides/hibernate-orm). My existing code is relying on the fact that transactions are using the EXTENDED persistence context type, however, quarkus-hibernate-orm seems to be always using PersistenceContextType.TRANSACTION and there is no way of configuring this. Thus, if I wanted to switch now, I would need to refactor a very large portion of the code.

**Comment by yrodiere** [comment]
Thanks for reporting this.

Feature-wise, it may be possible to have request-scoped entity managers that span multiple transactions, I worry a bit about the behavior if changes occur between transactions; unless you use version-based concurrency control, you might end up unwillingly erasing updates from concurrent transactions.
Simply put, this feels weird and dangerous, and I personally wouldn't use that. But I do understand legacy code exists. It would help if we had confirmation there are use cases for this in *new* code, though. Do you know of any?

@Sanne any opinion on extended persistence contexts? I vaguely remember this being discussed at some point, and I think you had an opinion, but I can't find this conversation...

**Comment by Treiblesschorle** [comment]
Thanks a lot for the quick answers.
> If you're really motivated perhaps you'd like to try implementing it?
Unfortunately, I don't have any good idea what this would entail, but for what you wrote it sounded like this is not trivial. How would you rate the complexity of this?

**Comment by yrodiere** [comment]
@Treiblesschorle I think you meant to close as not planned?

**Comment by Treiblesschorle** [comment]
Yes, thanks for changing it. I wanted to convey that this ticket is not relevant for me anymore, thus closed it. Did not pay enough attention to the resolution state, sorry.
